eventlistener/event_listener.go

```diff
@@ -63,8 +71,54 @@ func (e *EventListener) RegisterHandler(eventHash common.Hash, fn HandlerFn) {
 
 // Listen starts listening for events. To gracefully stop, call Stop().
 func (e *EventListener) Listen(ctx context.Context) error {
+	// Connect to all the Ethereum nodes.
+	var mapWsUrlToEthClient = make(map[string]*ethclient.Client)
```

Comment:
do you think this is better than running multiple instances?

Reply:
oh i was following the docs that you shared in node js 😅 

but we can have both, right? just run this and increase the instance size to 5 in cloud run for example

Reply:
I thought ethers' Fallback provider, not iterating from all nodes

Reply:
yeah, but for event listener we need to listen to all at the same time since we can't use "fallback" as we don't know when will the listener fail to listen?

if we deploy multiple instances it'll be the same too, we'll listen to multiple nodes at the same time. And it is harder to deploy since we need to create multiple deployments with different env file

Or do you have different implementation in mind wrt this?

Reply:
no I mean, you said you were following ether's Fallback provider
and I thought this is like translating ethers' implementation in golang

Reply:
we can follow the Ether's fallback provider when we implement for calling the RPC URL, not the WSS